### PR TITLE
Feature/doc secret store

### PR DESCRIPTION
**Pull request by alain2sf**
Secret Store documentation: creation, edition, formatting and redaction. Screenshots upload. Consul/Vault topics on Docker, launch, configuration, usage (CLI, API, UI).

### CODE REVIEW

**Comment by alain2sf** [comment]
@mhall119 @jpwhitemn @Ferriera Hey guys, just finished the Secret Store (Vault) documentation part, did this PR #1034. Cannot add reviewers, could you please check this out and approve/merge if OK. Thx.

**Comment by tsconn23** [comment]
Is this meant to target the delhi branch? Build is breaking due to the upstream influxdb dependency addressed in master via https://github.com/edgexfoundry/edgex-go/pull/1010. If this needs to go into the Delhi branch, then we will need an identical fix.

Also, please squash commits.

**Comment by alain2sf** [comment]
@tsconn23 Of course it's a Delhi branch, it's the security documentation for the Secret Store (Vault). Tingyu and me are finishing the work on the Security documentation. 
This PR is just the corresponding .rst file and a bunch of .png. How is it a problem to merge the PR into Delhi? I don't follow. 

I'm going to try to squash the commits while preserving the PR, never did this before, so I need a bit of learning. Do you have any advice?

**Comment by alain2sf** [comment]
@tsconn23 Just reading some Github tech notes and it seems you can use the “Squash and Merge” function in GitHub, basically git --squash merge branch. So, that would be more pragmatic to squash when you perform the merge. Let me know.

**Comment by tsconn23** [comment]
@alain2sf Squash commit [example](https://github.com/todotxt/todo.txt-android/wiki/Squash-All-Commits-Related-to-a-Single-Issue-into-a-Single-Commit)

Since documentation is in the same repo as code, committing changes will cause the verify jobs to run so the code has to be in a buildable state for your documentation to get merged. An upstream transitive dependency changed about 10 days ago and we have accounted for that in the `master` branch but not in `delhi` because no one was aware there would be more changes to this branch. You will need to wait for this PR to be merged after the above issue I referenced is fixed.

**Comment by tsconn23** [comment]
@alain2sf Please rebase your feature branch for this PR now that #1038 has been merged. Once you rebase and push, that will trigger the build verify jobs and I expect they will pass this time.

File: docs/security/Ch-SecretStore.rst [hunk 1]
```diff
@@ -2,6 +2,1057 @@
...
+
+    root@edgex-vault:/vault # vault secrets list
+    Path          Type         Accessor              Description
+    ----          ----         --------              -----------
+    cubbyhole/    cubbyhole    cubbyhole_ad070930    per-token private secret storage
+    identity/     identity     identity_5397dc2f     identity store
+    secret/       kv           kv_2362c227           key/value secret storage
+    sys/          system       system_410e4276       system endpoints used for control, policy and debugging
+
+.. note:: Line 5: EdgeX Foundry platform is using the Key/Value secret stroage named ``secret``
```
Comment (mhall119) [inline docs/security/Ch-SecretStore.rst]: Typo: "stroage" -> "storage"

File: docs/security/Ch-SecretStore.rst [hunk 2]
```diff
@@ -2,6 +2,1057 @@
...
+
+Vault can use seceral backend systems (filesystem, databases, Consul, Etcd, S3, Azure, etc.) to securely store every sensitive asset. The current EdgeX Foundry implementation of Vault is using `Consul <https://www.consul.io/>`_, another HashiCorp open source software product. Consul is a distributed service mesh to connect, secure, and configure services across any runtime platform and public or private cloud. Consul uses a consensus protocol to provide Consistency as defined by `CAP <https://en.wikipedia.org/wiki/CAP_theorem>`_. The consensus protocol is based on `"Raft: In search of an Understandable Consensus Algorithm" <https://ramcloud.stanford.edu/wiki/download/attachments/11370504/raft.pdf>`_. For a visual explanation of Raft, see The `Secret Lives of Data <http://thesecretlivesofdata.com/raft>`_.
```
[the remark this answers is not in the document]
Comment (alain2sf) [reply docs/security/Ch-SecretStore.rst]: fixed

File: docs/security/Ch-SecretStore.rst [hunk 3]
```diff
@@ -2,6 +2,1057 @@
...
+
+The seamless integration of Vault and Consul provides a strong yet simple infracstucture to setup a reliable high availability architecture (Vault failover nodes, Consul Clustering) for the EdgeX Foundry Security services in production.
```
[the remark this answers is not in the document]
Comment (alain2sf) [reply docs/security/Ch-SecretStore.rst]: fixed

File: docs/security/Ch-SecretStore.rst [hunk 4]
```diff
@@ -2,6 +2,1057 @@
...
+
+This Compose file starts the entire EdgeX Foundry platform including the security services. 
+
+The command to start EdgeX Foundry platform including the Secret Store and API gateway related services is:
+
+.. code-block:: shell
+
+    sh> docker-compose up -d
+
+For a pristine run, it is strongly recommended to thoroughly clean up any Docker artefacts remaining from the current run.
```
[the remark this answers is not in the document]
Comment (alain2sf) [reply docs/security/Ch-SecretStore.rst]: fixed

File: docs/security/Ch-SecretStore.rst [hunk 5]
```diff
@@ -2,6 +2,1057 @@
...
+
+    * **Leasing and Renewal:** All secrets in Vault have a lease associated with them (automatic revocation). However, clients can renew leases via built-in renew APIs.
+
+    * **Revocation:** Vault has built-in support for secret revocation. Single secrets, but also a tree of secrets. Revocation assists in key rolling as well as locking down systems in the case of an intrusion.
+
+======================
+Start the Secret Store
+======================
+
+Start the Secret Store with Docker Compose and a Docker Compose manifest file. The Docker Compose file named docker-compose-delhi-0.7.0.yml can be found at these two locations:
```
[the remark this answers is not in the document]
Comment (alain2sf) [reply docs/security/Ch-SecretStore.rst]: The 0.7.1 file is referenced in the API Gateway doc as well. I can test the security secret store service with the 0.7.1 base file but I will need @tingyuz to perform the same QA for the API Gateway. Then if everything is fine we can push the 0.7.1 file with security services, update the docs and push them as well.

File: docs/security/Ch-SecretStore.rst [hunk 6]
```diff
@@ -2,6 +2,1057 @@
...
+    edgex-vault        |
+
+
+.. note:: Line 4 & 7: Vault **API** endpoint on port 8200 (lines 4 and 7).
+
+          Line 7: Vault has **TLS** enabled.
+
+          Line 10: Vault backend storage is **Consul**.
+
+Start the fith service: vault-worker (Vault init/unseal process and setups)
```
[the remark this answers is not in the document]
Comment (alain2sf) [reply docs/security/Ch-SecretStore.rst]: fixed

File: docs/security/Ch-SecretStore.rst [hunk 7]
```diff
@@ -2,6 +2,1057 @@
...
+
+           Line 9: Shows the Consul nodes (1 node in EdgeX default configuration).
+
+           Lines 12-14: Show the Consul registered services.
+
+============================
+Configuring the Secret Store
+============================
+
+Vault server configuration is essentially concentrated in one JSON file named ``local.json``. This file was prepared during the Vault Docker image build process. In the eventuality of a change, the Vault server container should be accessed to then modify the JSON file. The absolute path being ``/vaul/config/local.json``. To reload the new configuration simply send Vault PID a HUP signal to trigger a configuration reload.
```
[the remark this answers is not in the document]
Comment (alain2sf) [reply docs/security/Ch-SecretStore.rst]: fixed

File: docs/security/Ch-SecretStore.rst [hunk 8]
```diff
@@ -2,6 +2,1057 @@
...
+
+    sh> docker exec -it -e PS1='\u@\h:\w \$ ' -e VAULT_CAPATH='/vault/config/pki/EdgeXFoundryCA/EdgeXFoundryCA.pem' edgex-vault sh
+
+    root@edgex-vault:/vault # ls -l
+    total 12
+    drwxr-xr-x    4 vault    vault         4096 Jan 13 13:34 config
+    drwxr-xr-x    2 vault    vault         4096 Jun  7  2018 file
+    drwxr-xr-x    2 vault    vault         4096 Jun  7  2018 logs
+
+Pay attention to the ``VAULT_CAPATH`` environment variable passed to the session. This is necessary in order to run succesfull Vault CLI command. Every Vault CLI command is a wrapper of the Vault HTTP API. The Vault server is configured with TLS using X.509 PKI materials generated and signed by a local self-signed CA (EdgeXFoundryCA). Therefore, in order for each Vault CLI command (or to that extent cURL commands) to verify the Vault server TLS certificate, the self-signing CA root certificate would have to be known by the CLI command interpreter. This ``VAULT_CAPATH`` variable is checked by every Vault CLI commands, alternatively each Vault CLI command can specify an option with the same certificate path if the variable is not set.
```
[the remark this answers is not in the document]
Comment (alain2sf) [reply docs/security/Ch-SecretStore.rst]: fixed

File: docs/security/Ch-SecretStore.rst (hunk 1 above)
Comment (alain2sf) [reply docs/security/Ch-SecretStore.rst]: fixed

**Comment by alain2sf** [comment]
> Mostly typos that need to be fixed.
> 
> This is quite a large single page document, breaking it up into multiple pages in the future will make it easier to read and maintain.

Yup, I put some effort in this one ;)  After update and QA of the compose file from 0.7.0 to 0.7.1, we'll need to update the various links/references then I'll break it up.

**Comment by alain2sf** [comment]
@mhall119 Thanks. I can't see the Delhi branch updated though, something's wrong ? @tsconn23 Can you confirm, thanks ?

**Comment by mhall119** [comment]
I don't typically merge PRs for edgex, but since many folks are at a face-to-face this week I've gone ahead and merged this one.

**Comment by alain2sf** [comment]
Thanks Michael, appreciate it.

**Comment by mhall119** [comment]
No problem. These should get published to the docs site in the next 24 hours.

**Pull request merged**